test/object-store/sync/app.cpp

```diff
@@ -1776,6 +1777,126 @@ TEST_CASE("app: set new embedded object", "[sync][app]") {
+        realm_config.sync_config->error_handler = [](std::shared_ptr<SyncSession>, SyncError error) {
+            std::cerr << error.message << std::endl;
+        };
+        realm_config.schema_version = 1;
+        realm_config.path = base_path + "/orig/default.realm";
+
+        std::mutex mutex;
+        auto task = realm::Realm::get_synchronized_realm(realm_config);
+        task->start([&](ThreadSafeReference ref, std::exception_ptr error) {
+            std::lock_guard<std::mutex> lock(mutex);
+            REQUIRE(!error);
+            realm_ref = std::move(ref);
+        });
+        util::EventLoop::main().run_until([&] {
+            std::lock_guard<std::mutex> lock(mutex);
+            return bool(realm_ref);
+        });
+        SharedRealm realm = Realm::get_shared_realm(std::move(realm_ref));
+
+        // Write some data
+        realm->begin_transaction();
+        CppContext c;
+        Object::create(c, realm, "Person",
+                       util::Any(realm::AnyDict{{"_id", util::Any(ObjectId::gen())},
+                                                {"age", INT64_C(64)},
+                                                {"firstName", std::string("Paul")},
+                                                {"lastName", std::string("McCartney")}}));
+        realm->commit_transaction();
+        wait_for_upload(*realm);
+        wait_for_download(*realm);
+
+        realm->write_copy(base_path + "/copy/default.realm", BinaryData());
+
+        // Write some additional data
+        realm->begin_transaction();
+        Object::create(c, realm, "Dog",
+                       util::Any(realm::AnyDict{{"_id", util::Any(ObjectId::gen())},
+                                                {"breed", std::string("stabyhoun")},
+                                                {"name", std::string("albert")},
+                                                {"realm_id", std::string("foo")}}));
+        realm->commit_transaction();
+        wait_for_upload(*realm);
+    }
+    // Starting a new session based on the copy
+    {
+        TestSyncManager sync_manager(TestSyncManager::Config(config), {});
+        auto app = sync_manager.app();
+        app->log_in_with_credentials(AppCredentials::anonymous(),
+                                     [&](std::shared_ptr<SyncUser> user, Optional<app::AppError> error) {
+                                         REQUIRE(!error);
+                                         REQUIRE(user);
+                                     });
+
+        ThreadSafeReference realm_ref;
+
+        realm::Realm::Config realm_config;
+        realm_config.sync_config = std::make_shared<realm::SyncConfig>(app->current_user(), bson::Bson("foo"));
+        realm_config.sync_config->client_resync_mode = ClientResyncMode::Manual;
+        realm_config.sync_config->error_handler = [](std::shared_ptr<SyncSession>, SyncError error) {
+            std::cerr << error.message << std::endl;
```

Comment:
I think we want to abort if we get here because that's how we'll know if the server sent us an error message, or am I missing something?

Reply:
Removed